batch/src/main/java/com/hartwig/batch/operations/rna/RnaIsofox.java

```diff
@@ -0,0 +1,171 @@
+    private static final String READ_LENGTH_76 = "76";
+    private static final String READ_LENGTH_151 = "151";
+
+    private static final double GC_BUCKET_76 = 0.0132;
+    private static final double GC_BUCKET_151 = 0.0066;
+    private static final int FRAG_LENGTH_FRAG_COUNT = 1000000;
+    private static final int LONG_FRAG_LENGTH_LIMIT = 550;
+    private static final String FRAG_LENGTH_BUCKETS = "50-0;75-0;100-0;125-0;150-0;200-0;250-0;300-0;550-0";
+    private static final String ENRICHED_GENE_IDS = " ENSG00000213741;ENSG00000258486;ENSG00000165525;ENSG00000266037;ENSG00000265150;ENSG00000264462;ENSG00000264063";
+
+    @Override
+    public VirtualMachineJobDefinition execute(
+            InputBundle inputs, RuntimeBucket bucket, BashStartupScript startupScript, RuntimeFiles executionFlags) {
+
+        InputFileDescriptor descriptor = inputs.get();
+
+        final String batchInputs = descriptor.value();
+        final String[] batchItems = batchInputs.split(",");
+
+        if(batchItems.length != 2)
+        {
+            System.out.print(String.format("invalid input arguments(%s) - expected SampleId,ReadLength", batchInputs));
+            return null;
+        }
+
+        final String sampleId = batchItems[0];
+        final String readLength = batchItems[1];
+
+        // copy down BAM and index file for this sample
+        final String bamFile = String.format("%s%s", sampleId, RNA_BAM_FILE_ID);
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp %s/%s/%s %s",
+                RNA_BAM_LOCATION, sampleId, bamFile, VmDirectories.INPUT));
+
+        final String bamIndexFile = String.format("%s%s", sampleId, RNA_BAM_INDEX_FILE_ID);
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp %s/%s/%s %s",
+                RNA_BAM_LOCATION, sampleId, bamIndexFile, VmDirectories.INPUT));
+
+        // copy down the executable
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp %s/%s %s",
+                ISOFOX_LOCATION, ISOFOX_JAR, VmDirectories.TOOLS));
+
+        startupScript.addCommand(() -> format("chmod a+x %s/%s", VmDirectories.TOOLS, ISOFOX_JAR));
+
+        // copy down required reference files
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp -r %s/%s %s",
+                ISOFOX_LOCATION, ENSEMBL_DATA_CACHE, VmDirectories.INPUT));
+
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp %s/%s %s",
+                ISOFOX_LOCATION, REF_GENOME, VmDirectories.INPUT));
+
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp %s/%s %s",
+                ISOFOX_LOCATION, REF_GENOME_INDEX, VmDirectories.INPUT));
+
+        final String expectedCountsFile = readLength.equals(READ_LENGTH_76) ? EXP_COUNTS_READ_76 : EXP_COUNTS_READ_151;
+
+        startupScript.addCommand(() -> format("gsutil -u hmf-crunch cp %s/%s %s",
+                ISOFOX_LOCATION, expectedCountsFile, VmDirectories.INPUT));
```

Comment:
Feels like all of these should either be pulled into the `InputBundle` (using the JSON input as mentioned above) or into the image.